src/apps/bridge/aanderaaController.cpp

```diff
@@ -0,0 +1,233 @@
+static constexpr uint32_t NODE_INFO_TIMEOUT_MS =            1000;
+static constexpr uint8_t AANDERAA_AVERAGER_MAX_SAMPLES =    10;
+
+static void createAanderaaSub(uint64_t node_id);
+static void runController(void *param);
+static Aanderaa_t *findAanderaaById(uint64_t node_id);
+static void sampleTimerCallback(TimerHandle_t timer);
+static bool node_info_reply_cb(bool ack, uint32_t msg_id, size_t service_strlen,
+                               const char *service, size_t reply_len, uint8_t *reply_data);
+
+bool Aanderaa::subscribe() {
+  bool rval = false;
+  char *sub = static_cast<char *>(pvPortMalloc(BM_TOPIC_MAX_LEN));
+  configASSERT(sub);
+  int topic_strlen = snprintf(sub, BM_TOPIC_MAX_LEN, "%" PRIx64 "%s", node_id, subtag);
+  if (topic_strlen > 0) {
+    rval = bm_sub_wl(sub, topic_strlen, aanderaSubCallback);
+  } 
+  vPortFree(sub);
+  return rval;
+}
+
+void Aanderaa::aanderaSubCallback(uint64_t node_id, const char *topic, uint16_t topic_len,
+                                  const uint8_t *data, uint16_t data_len, uint8_t type,
+                                  uint8_t version) {
+  (void)data;
+  (void)data_len;
+  (void)type;
+  (void)version;
+  printf("Aanderaa data received from node %" PRIx64 " On topic: %.*s\n", node_id, topic_len,
+         topic);
+  Aanderaa_t *aanderaa = findAanderaaById(node_id);
+  if (aanderaa) {
+    static AanderaaDataMsg::Data d;
+    if (AanderaaDataMsg::decode(d, data, data_len) == CborNoError) {
+      aanderaa->abs_speed_mean_cm_s.addSample(d.abs_speed_cm_s);
+      aanderaa->abs_speed_std_cm_s.addSample(d.abs_speed_cm_s);
+      aanderaa->direction_circ_mean_rad.addSample(degToRad(d.direction_deg_m));
+      aanderaa->direction_circ_std_rad.addSample(degToRad(d.direction_deg_m));
+      aanderaa->temp_mean_deg_c.addSample(d.temperature_deg_c);
+      if(aanderaa->abs_speed_mean_cm_s.getNumSamples() == AANDERAA_AVERAGER_MAX_SAMPLES) {
+        static char log_buf[256];
+        size_t log_buflen = snprintf(log_buf, sizeof(log_buf), "Aanderaa Node %" PRIx64 " Data: "
+                                                            "abs_speed_mean_cm_s: %.3f "
+                                                            "abs_speed_std_cm_s: %.3f "
+                                                            "direction_circ_mean_rad: %.3f "
+                                                            "direction_circ_std_rad: %.3f "
+                                                            "temp_mean_deg_c: %.3f\n",
+                                node_id, aanderaa->abs_speed_mean_cm_s.getMean(),
+                                aanderaa->abs_speed_std_cm_s.getStd(),
+                                aanderaa->direction_circ_mean_rad.getCircularMean(),
+                                aanderaa->direction_circ_std_rad.getCircularStd(),
+                                aanderaa->temp_mean_deg_c.getMean());
+        BRIDGE_LOG_PRINTN(log_buf, log_buflen);
```

Comment:
Right now just log the avg sample to the spotter rotated bridge log as a placeholder. Do fancier stuff later.